### PR TITLE
ci(deps): configure dependabot for github-actions

### PR DESCRIPTION
**Pull request by davorpa**
## What does this PR do?

1. *Dependabot checks for updates*. Dependabot pulls down your dependency files and looks for any outdated or insecure requirements.
2. *Dependabot opens pull requests*. If any of your dependencies are out-of-date, Dependabot opens individual pull requests to update each one.
3. *You review and merge*. You check that your tests pass, scan the included changelog and release notes, then hit merge with confidence.

So, **this PR configures [Dependabot](https://dependabot.com/) to automatize** this process for github-actions artifacts used in repository workflows.

As is config provided **needes add this labels** to github repo/org:
https://github.com/EbookFoundation/free-programming-books/blob/786ae8b00b09a418f9d70b6ea5f6c944bd813afd/.github/dependabot.yml#L17-L22

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Provide `dependabot.yml` config file.
- [x] Choose labels and add them to repo or if config is centralized, do it in organization.
- [ ] Enable dependabot in repo settings

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

**Comment by eshellman** [comment]
Would appreciate review from someone familiar with this.

**Comment by davorpa** [comment]
I'll know is a critical file of repo configuration, so I'll provide more info...

Dependabot aims to solve/fix security issues (something as is kwown as [CVE - Common Vulnerabilities and Exposures](https://en.wikipedia.org/wiki/Common_Vulnerabilities_and_Exposures)) making automatic pull request as way of recomendations. At those PR, an entire context with breaking changes, release/changelog history compatibility and so on will be provided as description. **In neither moment it commits nothing without supervision**.

:arrow_forward: A guide reference of how to activate or change proposed configuration can be started here:

https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/enabling-and-disabling-version-updates

:arrow_forward: On activate before merge this, a non parametrized file is suggested to commit, and be sure that conficts will appear.

:arrow_forward: I avoid to set assignees / reviewers parameters due to I see is a feature not used in PR process of this repo. Also schedule could be changed (now is at Central Europe on half day of Saturdays, perfect to handle also America timezone), or labels...

Let me know what is expected, and I'll change it. :wink: 

:arrow_forward: About all config parameters can be used in `dependabot.yml`:

https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates

:arrow_forward: This below list there are the dependencies that will be monitorized in both workflows, `fplint` and `urlbot`, we have:

- `actions/checkout`
- `actions/setup-node`
- `trilom/file-changes-action`
- `actions/upload-artifact`

I hope this helps to get in context :hugs:

**Comment by davorpa** [comment]
Today dependabot advice me about an action update

Here the example: https://github.com/davorpa/testing-github-actions/pull/1

**Comment by davorpa** [comment]
Today again another notification about github-actions comes into my inbox.

It cheet me that `setup-node` action should be upgraded.

https://github.com/davorpa/testing-github-actions/pull/3

**Comment by eshellman** [comment]
Question: on other repos I have, there's a bot that warns about security issues, but the alert and PRs are only visible to repo owner. Is that true for dependabot?

**Comment by davorpa** [comment]
> Question: on other repos I have, there's a bot that warns about security issues, but the alert and PRs are only visible to repo owner. Is that true for dependabot?

PRs are always public. Security alerts don't.

Moreover, I don't know if bot only response to reply commands (e.g.  `@dependabot merge and squash`) if you are a mantainer. I'd have to check it. Documentation don't tell us too much about.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Github Dependabot. Config file
```
Comment (CleanMachine1) [inline .github/dependabot.yml:1]:
```suggestion
# Github Dependabot config file
```

**Review by CleanMachine1 — COMMENTED** [review]
just a small nitpick

**Comment by LuigiImVector** [comment]
> Could someone try with [davorpa/testing-github-actions#2](https://github.com/davorpa/testing-github-actions/pull/2) that is still open?

Only who have the permissions can: https://github.com/davorpa/testing-github-actions/pull/2#issuecomment-1059770851

**Comment by davorpa** [comment]
> Only who have the permissions can: [davorpa/testing-github-actions#2 (comment)](https://github.com/davorpa/testing-github-actions/pull/2#issuecomment-1059770851)

Yep, I'll see it. thanks, Luigi.

Then I think it's safe enable this feature since is like a normal PR but with a bot commands and compatibility info as extra.

**Pull request merged**